### PR TITLE
Fix addRule for functional values, fix #475

### LINKED ISSUE #475: Need to detach sheet for addRule with function values
[Online Example](https://www.webpackbin.com/bins/-KiTXf9BfyDXWpbJ0i_J)

If we add new rule with just function values, `.update()` does not work for this new rule:

```js
const styles = {
  title: {
    fontSize: 20,
    color: ({color}) => color
  }
}

const sheet = jss.createStyleSheet(styles, {link: true}).attach()
sheet.addRule('subtitle', {color: ({color}) => color})
sheet.update({color: 'red'}) // still no color for subtitle, with attach() then too
// but if we call .deploy() - color applies
```

But this works:
```js
const sheet2 = jss.createStyleSheet(styles, {link: true}).attach()
sheet2.detach()
sheet2.addRule('subtitle', {color: ({color}) => color})
sheet2.update({color: 'red'}).attach() // red color for subtitle too
```

### PR DESCRIPTION
**Pull request by lttb**
The problem is that `rule.toString()` returns nothing if we have just function values in rule without calculated values. So it needs to return just an empty rule like `.class {}` to be applied to the stylesheet.

### CODE REVIEW

**Comment by kof** [comment]
Why do you call toString() before .update()?

**Comment by lttb** [comment]
We call .toString in injectRule, that called in addRulle, because we make .addRule before .update

**Comment by kof** [comment]
Maybe we should change the behavior there?

**Comment by kof** [comment]
I am not sure what will happen if we render a rule without props in different browsers.

**Comment by lttb** [comment]
But we can add rule without data (for lazy apply for example), and it's a normal behavior for user

**Comment by kof** [comment]
Yeah, need to run tests in all browsers.

**Comment by lttb** [comment]
if we will have some issues with browsers, we always can render an empty prop

**Comment by kof** [comment]
what is an "empty" prop?

**Comment by lttb** [comment]
> what is an "empty" prop?

just an empty string, for example `.class {color:''}`
or we can also make it `initial`

**Comment by kof** [comment]
color: '' is an invalid value, I don't know what this might cause.

**Comment by kof** [comment]
"initial" is an actual value which might lead to unexpected results from the user perspective

**Comment by iamstarkov** [comment]
`color`'s `initial` is different in each browser

File: tests/functional/sheet.js [hunk 1]
```diff
@@ -247,6 +247,36 @@ describe('Functional: sheet', () => {
     })
   })
 
+  /*
```
Comment (kof) [inline tests/functional/sheet.js]: why is this commented out?

File: src/utils/toCss.js [hunk 2]
```diff
@@ -47,17 +47,20 @@ export default function toCss(selector: string, style: JssStyle, options: Option
...
       value = style[`$${prop}`]
+      hasFunctionValue = true
     }
     if (value != null && prop !== 'fallbacks') {
       result += `\n${indentStr(`${prop}: ${toCssValue(value)};`, indent)}`
     }
   }
 
-  if (!result) return result
+  if (!(result || hasFunctionValue)) return result
```
Comment (kof) [inline src/utils/toCss.js]: much more readable:

```js
if (!result && !hasFunctionValue) return result
```

File: tests/functional/sheet.js (hunk 1 above)
Comment (lttb) [inline tests/functional/sheet.js]: sorry, just forgot to uncomment

**Comment by kof** [comment]
some browsers time out on browserstack, trying to figure out why.
